src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,272 @@
+            NoSuchAlgorithmException, NoSuchProviderException {
+        X500Name x500Name = new X500Name("cn=SAML-jenkins");
+        X509Certificate[] chain = new X509Certificate[1];
+        chain[0] = getSelfCertificate(x500Name, new java.util.Date(), KEY_VALIDITY, keypair);
+        return chain;
+    }
+
+    /**
+     * Create a new keystore.
+     *
+     * @param keystore the keystore object.
+     * @param password the password to set to the keystore.
+     * @return the new keystore.
+     * @throws KeyStoreException        @see KeyStoreException
+     * @throws IOException              @see IOException
+     * @throws NoSuchAlgorithmException @see NoSuchAlgorithmException
+     * @throws CertificateException     @see CertificateException
+     */
+    private KeyStore initKeyStore(File keystore, String password)
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+        ks.load(null, password.toCharArray());
+        saveKeyStore(keystore, ks, password);
+        return ks;
+    }
+
+    /**
+     * save the keystore to disk.
+     *
+     * @param keystore file to save the keystore.
+     * @param ks       the keystore object.
+     * @param password the password to set to the keystore.
+     * @throws KeyStoreException        @see KeyStoreException
+     * @throws IOException              @see IOException
+     * @throws NoSuchAlgorithmException @see NoSuchAlgorithmException
+     * @throws CertificateException     @see CertificateException
+     */
+    private void saveKeyStore(File keystore, KeyStore ks, String password)
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        FileOutputStream fos;
+        fos = new FileOutputStream(keystore);
+        try {
+            ks.store(fos, password.toCharArray());
+        } finally {
+            fos.close();
+        }
+    }
+
+    /**
+     * load a keystore from a file. if it fails create a new keystore.
+     *
+     * @param keystore path to the keystore.
+     * @param password password of the keystore.
+     * @return the keystore loaded.
+     * @throws KeyStoreException        @see KeyStoreException
+     * @throws IOException              @see IOException
+     * @throws CertificateException     @see CertificateException
+     * @throws NoSuchAlgorithmException @see NoSuchAlgorithmException
+     */
+    private KeyStore loadKeyStore(File keystore, String password)
+            throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+        try (InputStream in = new FileInputStream(keystore)) {
+            ks.load(in, password.toCharArray());
+        } catch (Exception e) { // better to catch something more specific
```

Comment:
`better to catch something more specific`

Well, I left the comment there for you to fix it 😄 
Don't catch `Exception`, catch whatever is thrown by the `try` block.